src/goPackages.ts

```diff
@@ -66,23 +82,27 @@ function gopkgs(): Promise<Map<string, string>> {
 	});
 }
 
-function getAllPackagesNoCache(): Promise<Map<string, string>> {
+function getAllPackagesNoCache(workDir?: string): Promise<Map<string, string>> {
 	return new Promise<Map<string, string>>((resolve, reject) => {
 		// Use subscription style to guard costly/long running invocation
 		let callback = function (pkgMap: Map<string, string>) {
 			resolve(pkgMap);
 		};
-		gopkgsSubscriptions.push(callback);
 
-		// Ensure only single gokpgs running
-		if (!gopkgsRunning) {
-			gopkgsRunning = true;
-			gopkgs().then((pkgMap) => {
-				gopkgsRunning = false;
+		let subs = gopkgsSubscriptions.get(workDir);
+		if (!subs) {
+			subs = [];
+			gopkgsSubscriptions.set(workDir, subs);
+		}
+		subs.push(callback);
 
-				let subs = gopkgsSubscriptions;
-				gopkgsSubscriptions = [];
+		// Ensure only single gokpgs running
+		if (!gopkgsRunning.has(workDir)) {
+			gopkgsRunning.add(workDir);
 
+			gopkgs(workDir).then((pkgMap) => {
+				gopkgsRunning.delete(workDir);
```

Comment:
Entries never get deleted from `gopkgsRunning` when `gopkgs` returns a rejected promise

Reply:
No rejection on gopkgs. Positive and negative result will be return by `resolve()`.

Base on your comment, negative should be in form of rejection. Such as:
1. When `gopkgs` binary not found
2. When `gopkgs` needs to update

This is part of quality I think, which I'm okay, we can do this. But maybe after this PR.